bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/discovery/HueBridgeDiscoveryParticipant.java

```diff
@@ -92,4 +92,10 @@
         }
         return null;
     }
+
+    @Override
+    public int getRemovalGracePeriodSeconds(RemoteDevice device) {
+        // Hue bridges have maxAge 100 seconds, so apply a grace period of half that
+        return 50;
```

Comment:
If we don't know for sure if this value is working we might want to add a dynamic configuration property to adopt the value during runtime (e.g. by adding `discovery.hue:gracePeriod=100` to the `$OPENHAB_CONF/services/runtime.cfg`)?

Reply:
Good point. I will add a Configuration Parameter with a default value. Probably it needs some new text in ReadMe as well. I will do that.

Reply:
You could define a Hue binding (not thing) configuration setting ?

Reply:
Another option which came to my mind is to implement a switch to disabled the UPnP discovery for the Hue binding completely. I did the same thing for Kodi and avmfritz. You can have a look over there how it is working.

Reply:
I am rtalking about that (example for the Netatmo binding):
![image](https://user-images.githubusercontent.com/10186704/106355425-0b8af680-62f8-11eb-8de1-1a62f15cb6ef.png)

Unfortunately, the access to these settings is now very natural in MainUI....

Reply:
The setting must be defined in the binding.xml file. You can find examples in few bindings (netatmo or sonos for example).

Reply:
> define a Hue binding (not thing) configuration setting 

Of course, you are right that it cannot be a Configuration Parameter of the Bridge Thing since it is not yet instantiated at this point.

> implement a switch to disabled the UPnP discovery

Or you could modify those bindings to implement the gracePeriod :) ??

Reply:
This will not be so obvious as I imagined as the discovery is a service. I am not sure how easy it is to get the binding properties.
Th esolution from @cweitkamp is certainly easier, that is having the setting as a parameter of the service. But in this case, th esetting will probably not be adjustable in MainUI.

Reply:
@lolodomo We have to look at those settings (Netatmo, Sonos) from a different point of view. They are handling a discovery service attached to the BridgeHandler implementing an AbstractDiscoveryService. The UPnP based discovery is an own OSGi component without a relationship to the handler or HandlerFactory at all.

Reply:
:) Look at my previous message, I realized that ;)

Reply:
With your solution with a service parameter, will this automatically add a new entry in this part of MainUI ?
![image](https://user-images.githubusercontent.com/10186704/106355913-873a7280-62fb-11eb-9b78-191a58296dd6.png)

Reply:
No, it won't.

Reply:
:( So it will require a good entry in the documentation to explain how to adjust this setting.

Reply:
I just pushed a new commit as follows..
- Defines the new config parameter in 'binding.xml' with a default value
- Loads it in 'HueBridgeDiscoveryParticipant`

Note: I am by far not an expert in the OSGI stuff, but used the same `configurationPid = "binding.hue"` tag on the 'HueBridgeDiscoveryParticipant` component as is already used on the `HueBridgeHandlerFactory` component. My reason was as follows..

- Using it in `HueBridgeHandlerFactory` should allow the configuration parameter to be accessed and edited in MainUI.
- Using it in 'HueBridgeDiscoveryParticipant` should allow it to be accesses by the discovery participant.

**NOTE: I need your help in fully testing this**: I could not find a way to integrate a) the modified [discovery.upnp] interface, b) the modified [hue] binding, and c) the MainUI all on one system. The problem is that if I build the binding and the core module locally, I can only test them by dropping them in the addons folder. But if I do that then the binding's Configuration page is not accessible in MainUI (MainUI only shows those bindings that is has downloaded and installed itself).

Reply:
^ 
correction I meant `HueThingHandlerFactory` above..

Reply:
@cweitkamp @lolodomo any thought about my 'hack' of using the same 'configurationPid' on the handler factory and on the discovery participant? As far as I can tell, it seems to work insofar as the UI can see and edit the parameter, and the discovery participant can retrieve it. But I just wonder if this might break something on the discovery participant side?

```
@Component(service = UpnpDiscoveryParticipant.class, configurationPid = "binding.hue")
```

Reply:
I was asking myself the same question. 
Looks like a hack ... but a clever one :)

Reply:
I did some Googling and found [this article](https://stackoverflow.com/questions/55260518/how-to-use-a-single-config-file-for-multiple-osgi-services) -- there are two answers but neither is definitive: I used the second answer ("duplicate 'configurationPid's aren't forbidden", but I could adopt the first one if you prefer. => Please advise..

Reply:
I decided adopt the first answer from the [article](https://stackoverflow.com/questions/55260518/how-to-use-a-single-config-file-for-multiple-osgi-services) after all. I just pushed a new commit. It is IMHO much cleaner :)

Reply:
> I could not find a way to integrate a) the modified [discovery.upnp] interface, b) the modified [hue] binding, and c) the MainUI all on one system

I solved this. It is true that MainUI won't offer configuration for bindings that it has not installed, but it does offer it when you try to scan for Things manually.

![image](https://user-images.githubusercontent.com/893994/106482727-a4ab4000-64a5-11eb-8347-2f9ffa6ea3b2.png)

Reply:
> it will require a good entry in the documentation 

Also done :)

Reply:
And does it work? ;) I mean is your variable really set to the value in your configuration setting?

Reply:
> is your variable really set to the value in your configuration setting?

It is indeed! Although I say it myself, I think it is a rather cool new feature :)

(..you can test it yourself by dropping the two jars from this PR and https://github.com/openhab/openhab-core/pull/2144 into you addons folder..)